spec/jobs/descmetadata_download_job_spec.rb

```diff
@@ -18,21 +18,20 @@
   end
   let(:item1) { Dor::Item.new }
   let(:item2) { Dor::Item.new }
-  let(:object_client1) { instance_double(Dor::Services::Client::Object, find: cocina_model1) }
-  let(:object_client2) { instance_double(Dor::Services::Client::Object, find: cocina_model2) }
+  let(:object_client1) { instance_double(Dor::Services::Client::Object, find: cocina_model1, metadata: metadata_client1) }
+  let(:object_client2) { instance_double(Dor::Services::Client::Object, find: cocina_model2, metadata: metadata_client2) }
+  let(:metadata_client1) { instance_double(Dor::Services::Client::Metadata, mods: item1.descMetadata.content) }
+  let(:metadata_client2) { instance_double(Dor::Services::Client::Metadata, mods: item2.descMetadata.content) }
   let(:cocina_model1) { instance_double(Cocina::Models::DRO) }
   let(:cocina_model2) { instance_double(Cocina::Models::DRO) }
 
   before do
     allow(Dor::Services::Client).to receive(:object).with(pid_list[0]).and_return(object_client1)
     allow(Dor::Services::Client).to receive(:object).with(pid_list[1]).and_return(object_client2)
-
-    allow(Dor).to receive(:find).with(pid_list[0]).and_return(item1)
-    allow(Dor).to receive(:find).with(pid_list[1]).and_return(item2)
   end
 
   after do
-    FileUtils.rm_rf(output_directory) if Dir.exist?(output_directory)
+    FileUtils.rm_rf(output_directory)
```

Comment:
No need to check if dir exists when using `rm_rf` as this method will not complain if the dir doesn't exist. Prefer asserting over asking.